File: HYPERLEDGER/bfn-network-apps/businesslibrary/lib/data/customer.dart
import 'package:businesslibrary/data/misc_data.dart';

class Customer extends BaseParticipant {
  String participantId;
  String name;
  String cellphone;
  String email;
  String description;
  String address, dateRegistered;
  String country;
  bool allowAutoAccept;

  Customer(
      {this.participantId,
      this.name,
      this.cellphone,
      this.email,
      this.description,
      this.address,
      this.country,
      this.allowAutoAccept,
      this.dateRegistered});

  static const National = "NATIONAL",
      Municipality = 'MUNICIPALITY',
      Provincial = 'PROVINCIAL';

  Customer.fromJson(Map data) {
    print(data);
    this.participantId = data['participantId'];
    this.name = data['name'];
    this.description = data['description'];
    this.cellphone = data['cellphone'];
    this.address = data['address'];
    this.email = data['email'];
    this.country = data['country'];
    this.dateRegistered = data['dateRegistered'];
    this.allowAutoAccept = data['allowAutoAccept'];
  }

  Map<String, dynamic> toJson() => <String, dynamic>{
        'participantId': participantId,
        'name': name,
        'description': description,
        'cellphone': cellphone,
        'address': address,
        'email': email,
        'country': country,
        'dateRegistered': dateRegistered,
        'allowAutoAccept': allowAutoAccept,
      };
}

File: HYPERLEDGER/bfn-network-chaincode/bfn-chaincode2/src/helpers/models/customer.ts

import { JsonObject, JsonProperty } from 'json2typescript';

@JsonObject('Customer')
export class Customer {

  @JsonProperty('docType', String)
  public docType?: string;
  @JsonProperty('participantId', String)
  public participantId: string;
  @JsonProperty('name', String)
  public name: string | undefined;
  @JsonProperty('cellphone', String)
  public cellphone: string;
  @JsonProperty('email', String)
  public email: string | undefined;
  @JsonProperty('documentReference', String)
  public documentReference: string;
  @JsonProperty('description', String)
  public description: string;
  @JsonProperty('address', String)
  public address: string;
  @JsonProperty('dateRegistered', String)
  public dateRegistered: string;
  @JsonProperty('sector', String)
  public sector: string;
  @JsonProperty('country', String)
  public country: string | undefined;
}
